python-lib/dku_timeseries/interval_restriction.py

```diff
@@ -186,3 +186,22 @@ def _detect_segment(self, df, datetime_column, filter_column, filter_function, d
             segment_df = pd.DataFrame(columns=df_copy.columns)
 
         return segment_df.rename_axis(datetime_column).reset_index()
+
+    def _initialize_edges(self, df):
+        """
+        Adds an empty row at the beginning and the end of the dataset to initialize the algorithm
+        """
+        df_copy = df.copy()
+        if len(df) > 0:
+            columns_without_date = df_copy.columns[1:]
```

Comment:
When I test, the date column is not in df_copy.columns

Reply:
You're right, thank you for double checking that! I fixed it here : https://github.com/dataiku/dss-plugin-timeseries-preparation/pull/22/commits/c9d06f03e1ec32f13e886bc0efe3fe85e82c276f